### PR TITLE
Hts 1592 Find enrolment and Get email mongo queries ignore nino suffix

### PR DESCRIPTION
**Pull request by Jackie1988**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosave/repo/EmailStore.scala [hunk 1]
```diff
@@ -132,13 +136,12 @@ class MongoEmailStore @Inject() (mongo:   ReactiveMongoComponent,
...
   }
 
-  private[repo] def doUpdate(encryptedEmail: String, nino: NINO)(implicit ec: ExecutionContext): Future[Option[EmailData]] =
-    collection.findAndUpdate(
-      BSONDocument("nino" -> nino),
-      BSONDocument("$set" -> BSONDocument("email" -> encryptedEmail)),
-      fetchNewObject = true,
-      upsert         = true
-    ).map(_.result[EmailData])
+  private[repo] def doUpdate(encryptedEmail: String, nino: NINO)(implicit ec: ExecutionContext): Future[Boolean] =
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosave/repo/EmailStore.scala]: would we need to delete by regex when deleting emails as well?

File: app/uk/gov/hmrc/helptosave/repo/EnrolmentStore.scala [hunk 2]
```diff
@@ -117,7 +117,9 @@ class MongoEnrolmentStore @Inject() (mongo:   ReactiveMongoComponent,
     {
       val timerContext = metrics.enrolmentStoreGetTimer.time()
 
-      find("nino" → JsString(nino)).map { res ⇒
+      val regex = "^" + nino.take(8) + ".$"
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosave/repo/EnrolmentStore.scala]: would we need to update by regex when updating the enrolment store?

File: app/uk/gov/hmrc/helptosave/repo/EmailStore.scala (hunk 1 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosave/repo/EmailStore.scala]: Yes as its possible it won't be picked up if a different suffix is used

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**